RFC-002-SWAP.md

```diff
@@ -10,267 +10,255 @@
-    },
-    "beta_ledger": {
-      "value": "bitcoin",
-      "parameters": { "network": "mainnet" }
-    },
-    "alpha_asset": {
-      "value": "erc20",
-      "parameters": {
-        "quantity": "42000000000000000000",
-        "address": "0xB97048628DB6B661D4C2aA833e95Dbe1A905B280"
-      }
-    },
-    "beta_asset": {
-      "value": "bitcoin",
-      "parameters": { "quantity": "10000" }
+  "type": "RESPONSE",
+  "id": 0,
+  "payload": {
+    "headers": {
+      "decision": "accepted"
     },
-    "protocol": "comit-rfc-003",
-  },
-  "body": {},
+    "body": { ... },
+  }
 }
 ```
 
-## SWAP RESPONSE
-
-### Request accepted
+#### Declined SWAP REQUEST
 
 ```json
 {
-  "status": "OK20",
-  "headers": {},
-  "body": {},
+  "type": "RESPONSE",
+  "id": 0,
+  "payload": {
+    "headers": {
+      "decision": "declined"
+    },
+    "body": { ... },
+  }
 }
 ```
 
-### Request declined due to non-beneficial rate with hints
+## Registry extensions
 
-```json
-{
-  "status": "RE20",
-  "headers": {
-    "reason": {
-      "value": "rate-declined",
-      "parameters": {
-        "alpha_asset": {
-          "value": "erc20",
-          "parameters": {
-            "quantity": "42500000000000000000",
-            "address": "0xB97048628DB6B661D4C2aA833e95Dbe1A905B280"
-          }
-        },
-        "beta_asset": {
-          "value": "bitcoin",
-          "parameters": { "quantity": "10000" }
-        },
-      },
-    },
-  },
-  "body": {},
-}
-```
+This RFC extends the registry with the following elements:
 
-### Request rejected without a reason
+### The type `Ledger`
 
-```json
-{
-  "status": "RE21",
-  "headers": {},
-  "body": {},
-}
-```
+A section "Ledgers" is added to the registry which tracks all currently defined ledger types.
+Subsequent RFCs can refer to this type if they want to define a particular ledger.
+
+### The type `Asset`
+
+A section "Assets" is added to the registry which tracks all currently defined asset types.
+Subsequent RFCs can refer to this type if they want to define a particular asset.
+
+### The type `SwapProtocol`
+
+A section "Protocols" is added to the registry which tracks all currently defined protocols.
+Subsequent RFCs can refer to this type if they want to define a particular swap protocol.
+
+### The type `DeclineBody` and the initial set of possible values
+
+A section "DeclineBodys" is added to the registry which tracks all currently defined reasons.
+Subsequent RFCs can refer to this type if they want to define new reasons for declining SWAP REQUESTs.
+
+The following `DeclineBody`s are added to the list.
+Each heading represents a `reason`.
+
+#### `unsatisfactory-rate`
+
+The rate of `alpha_asset` to `beta_asset` is not satisfactory to the receiver.
+
+##### `details`
+
+TBD 
+
+#### `protocol-unsupported`
+
+The protocol specified in the `protocol` header is not known to the receiving party.
+
+##### `details`
+
+TBD <!-- List known protocols in details -->
+
+#### `unknown-ledger`
```

Comment:
Interesting idea.

`unknown-ledger` could be something that is generated by a deserializer (hence automatic) whereas `unsupported` would mean a configuration somewhere.

I am leaning towards having that separated because (1) means there is no point in retrying and (2) might be successful later.

Reply:
For me this was more of a semantical issue, i.e. `unsupported-ledger` implies `unknown-ledger` while the other way around does not hold, e.g. we could have a generic ledger implementation which can handle any ledger (🕺)

Reply:
The separation sounds good but I do not see any `unsupported-*` in the file. Please add it.